### DISCUSSION TITLE
A new variant where pieces leave trails

### DISCUSSION THREAD

**Discussion by lykahb** [fairy-stockfish/Fairy-Stockfish#578] [Show and tell]
Hi, I've made a playable variant of chess https://www.lykahb.com/trailchess/. The game starts on an empty board. The players drop and move the pieces.

The game is playable and uses the same board library as lichess. To try it, play for both players or choose AI for your opponent. If you choose AI for two players, it'd autoplay. The AI is very simple, though.

I'd be interested to add support for it to Fairy-Stockfish.

It looks like there are two challenges here:
- How to add the concept of trails.
  - Pieces leave trails that need to be tracked.
  - A move now may be accompanied by the choice of trail both for the piece that moved (knight may leave one of L-shaped trails), and the choice of part of the trail that was cut. The move notation would need to be updated.
- The win condition that a player does not have a legal move does not exist in variants.ini

Can you share some advice on how to approach this?

**Comment by ianfab** [comment]
Hi. Having no legal move is equivalent to stalemate in chess terms and that is configurable, so that part should be trivial. However, the concept of trails seems vastly different from game dynamics in other variants, so spontaneously I do not have a good idea how this could be done.
